### PR TITLE
feat(workspace): Add workspaces to specify build order dependencies in packages.

### PR DESCRIPTION
**Pull request by jafin**
## Workspaces

- Add in npm workspaces to provide order to build dependencies.
- Move `oidc-service-worker` up a level outside of packages/react
- Update workflows with changes to workspaces.
- Adjust `test` for each project to be non-watch mode (Added test:watch as replacement). So that pipeline can call test from root and all packages can run test for CI
- cpy introduced as copyfiles I was having issues with it copying foldernames instead of folder contents. 

🐲🐲🐲 NOTE: I have not tested the changes to the workflows, it was a *BEST EFFORT* approach and will need review.

This should assist in easing the package dependency chain. 👏

### Decisions

I looked into `lerna` and installed it, but it wasn't doing much above what native npm workspaces added.  Lerna documentation also notes that they now lean on npm(pnpm/yarn) workspaces (something lerna->bootstrap used to provide).  Perhaps there are some other benefits I didn't discover?

Lerna can still be added, it should be a simple update to add it into the root project without much change, and just the various npm workspace commands with lerna equivalent.

### CODE REVIEW

**Comment by guillaume-chervet** [comment]
Thank you again @jafin , that awesome.
learna just automate publishing many packages dependencies. But for the moment this repo is not big enough for it.

Just give me some time to learn abou npm workspace :p

Do you have a twitter account or a linkedin?

**Comment by jafin** [comment]
@guillaume-chervet Sorry I don't have an active Twitter/linked in at the moment.

With respect to publishing, you may also be interested in looking at [changeset](https://github.com/changesets/changesets).  When I tested things on my fork of this repo, I setup a github workflow and found it quite easy to perform publishing to npm registry.

One thing I did like with it, was the way you can explicitly trigger a release (still from GitHub web), to allow queueing a set of changes, avoiding minor point releases for `every commit` to main.  

But this would require a change in PR's, to request a user to run a changeset command to enter a description of the change, and if it was a minor/major breaking change etc.

I had a example github workflow here https://github.com/jafin/react-oidc/blob/master/.GitHub/workflows/release.yml It was quite basic though.

**Comment by guillaume-chervet** [comment]
Very Sorry @jafin  for the delay, I'am lacking of times these days .

**Comment by jafin** [comment]
@guillaume-chervet is all fine, I will need to rework this anyhow after the repo changes, it will be a little while for me to revisit. Changing to draft.

**Comment by guillaume-chervet** [comment]
hi @jafin , do you  want to continue your pullrequest ?
I'am started to really need this :)

**Comment by jafin** [comment]
Hi @guillaume-chervet 
Apologies for leaving this hanging.

I can have a look this weekend.  

Some questions for on approach. 

As I said above I tried lerna and reverts to npm workspaces as I wasn't seeing a huge benefit from having LERNA.
So I would still perhaps suggest going for npm + workspaces or even pnpm + workspaces.?

Let me know.

**Comment by guillaume-chervet** [comment]
Hi @jafin ,

I have read more documentation.
npm workspace is perfect.

New lerna version documentation say that learna is compatible with npm workspace.
Learna is not necessary for now. It is just about synchronize packages version for publishing packages. It can be bring after.

Thank you so much for you help.

A little more roadmap information for you. I aim to:
- add workspaces
- rename vanilla-oidc to oidc-client
- make react-oidc depends on oidc-client
- upgrade documentation to explain more about oidc-client
- rename repository to oidc-client
- continue clean code and unit test
- start exchange for openid certification

**Comment by jafin** [comment]
@guillaume-chervet sounds good!
I'll fix this PR up in the next few days for review.

**Comment by guillaume-chervet** [comment]
That look awesome @jafin

**Comment by jafin** [comment]
Hi @guillaume-chervet  This turned out bigger than I though.
Would you be able to have a look at it again?  I think it should be a in a good position to be reviewed.

I have not done anything in relation to packaging and versioning, so I am unsure if the pipeline scripts need work/amendment.
The workspace dependencies currently use workspace:* references which should get replaced on pack/publish, but i haven't done any checks here yet.

Ideally, it would be nice if this project could publish betas or alphas as to avoid end users being interrupted in the event there are errors.  (I have not done anything in regard to versioning with this change)

## todo
- [ ] Packaging libraries and ensure versioning of dependencies is correctly set. (instead of workspace) (@guillaume-chervet ?
- [ ] react-vite-demo is now almost redundant with react-oidc-demo (which uses vite) DELETE?
- [ ] You mentioned to rename oidc-vanilla,  I have renamed the package folder, but the package name is the same still. I did not want to make that change now as there were already too many things in this PR.

## Changes
- rename vanilla-oidc to oidc-client (did not change pkg name though)
- move react/src/oidc/vanilla -> oidc-client/src/
- export `getParseQueryStringFromLocation` , `getPath` , `getFetchDefault` `TokenRenewMode` from vanilla.
- jsxnext:main replaced with module (https://www.reddit.com/r/javascript/comments/5jwg9c/confused_about_fields_module_and_jsnextmain_in/) in `package.json`
- replace copyfiles with cpy (just to standardise, and cpy seems the dominant used pkg)
- separate the react-oidc-demo code moved to examples/react-oidc-demo/ from the react-oidc client package.
- migrate cra apps to vite
- add a basic build/test CI pipeline, it could run on each PR/commit as a lightweight check.
- Update to typescript 5.  Change some of the exports to `export type`

## Issues Encountered

### Running the demo but ran `pnpm i` from root, will make it miss the service-worker
 
If you run `pnpm i` from the root, then the postinstall sripts treat the execution path, root, as the destination to copy script to /public.  Which then stops the demo having the service worker scripts in the correct place.

The work around, is to run pnpm i from the demo project folder, and the postinstall script will copy the file correctly.  Perhaps this can be improved in future.

**Comment by guillaume-chervet** [comment]
So awesome @jafin , i test it tomorrow and i will prepare a branch that can publish alpha version to test it.

File: packages/react-oidc/package.json [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+    "prepare": "pnpm run clean && pnpm run copy-service-worker && pnpm run build",
+    "lint": "eslint src"
+  },
+  "peerDependencies": {
+    "react": "^17.0.0 || ^18.0.0",
+    "react-dom": "^17.0.0 || ^18.0.0",
+    "react-router-dom": "^6.0.0"
+  },
+  "dependencies": {
+    "cpy": "^10.1.0",
```
Comment (guillaume-chervet) [inline packages/react-oidc/package.json:48]: These 2 librairies lookks like DevDependencies

File: packages/react-oidc/package.json [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+    "cpy-cli": "^5.0.0"
+  },
+  "devDependencies": {
+    "@axa-fr/oidc-client-service-worker": "workspace:*",
+    "@axa-fr/vanilla-oidc": "workspace:*",
```
Comment (guillaume-chervet) [inline packages/react-oidc/package.json:53]: I think this one is more a direct "dependencies"

**Comment by guillaume-chervet** [comment]
Thank you so much @jafin . 
This PullRequest is a huge Job.

I have created a alpha branch. The idea is that a merge to alpha branch will publish an alpha automaticaly.

**Comment by jafin** [comment]
> Thank you so much @jafin . This PullRequest is a huge and excelent Job. I have created a alpha branch. The idea is that a merge to alpha branch will publish an alpha automaticaly.
> 
> I have tested demo and they all works !

THank you @guillaume-chervet ,  I think having the option to release test builds before new official version may ensure bugs are fixed before a new official version rolls out.  Just a comment, the way you have it setup now in this repo, aligns with github-flow, single branch for deploys makes it simple and quick to do always ready releases.  I didn't want to complicate that with other branches, sorry for the misunderstanding.  Just a way to push to main without a release being created,  perhaps if the CI is able to release a beta via tagging rather than always create new release on push to PR, that would be great, and avoid having unnecessary branches.  And still adhering to the github-flow apporach

Thank you for reviewing the changes. I'll update the changes you mentioned above and notify when done.

**Comment by guillaume-chervet** [comment]
Thank you again @jafin , you force me to find a better solution.

I do not like tag because it is too manual. Too many actions.

Here an idea of a solution, continue to use last git commit comment and automate everything.
- feat(oidc): my message (alpha) => will publish next minor as an alpha
- feat(oidc): my message (beta) => will publish next minor as an beta
- feat(oidc): my message (release)  => will publish next minor release 

With the script bellow :
 
```yaml
# This workflow will run tests using node and then publish a package to GitHub Packages when a release is created
# For more information see: https://help.github.com/actions/language-and-framework-guides/publishing-nodejs-packages

name: React Oidc CI

on:
  push:
    branches: [ main ]
  pull_request:
    branches: [ main ]

jobs:
  skip_ci:
    runs-on: ubuntu-latest
    outputs:
      canSkip: ${{ steps.check.outputs.canSkip }}
    steps:
      - id: check
        uses: Legorooj/skip-ci@main
  tests:
    runs-on: ubuntu-latest
    if: needs.skip_ci.outputs.canSkip != 'true' && github.event.pull_request.head.repo.fork
    steps:
      - uses: actions/checkout@v3
      - uses: actions/setup-node@v2
        with:
          node-version: 18

      - uses: pnpm/action-setup@v2
        name: Install pnpm
        with:
          version: 8.5.1
          run_install: false

      - name: pnpm i
        run: pnpm i --frozen-lockfile
        working-directory: ./packages/react-oidc

      - name: pnpm prepare
        run: pnpm run prepare
        working-directory: ./packages/react-oidc

      - name: pnpm test
        run: pnpm test -- --run
        working-directory: ./packages/react-oidc

      - name: pnpm install
        run: pnpm i --frozen-lockfile
        working-directory: ./packages/oidc-client

      - name: pnpm prepare
        run: pnpm run prepare
        working-directory: ./packages/oidc-client

  build:
    environment: react-oidc
    runs-on: ubuntu-latest
    if: needs.skip_ci.outputs.canSkip != 'true' && !github.event.pull_request.head.repo.fork
    steps:
      - uses: actions/checkout@v3
        with:
          token: ${{ secrets.GIT_TOKEN }}
      - name: Determine Alpha, Beta or Release
        id: which_tag
        run: |
          last_commit_message=$(git log --format=%B -n 1)
          # Check if last comment and with "(alpha)"
          if [[ $last_commit_message == *"(alpha)" ]]; then
            echo "tag=alpha" >> $GITHUB_OUTPUT
          fi
          if [[ $last_commit_message == *"(beta)" ]]; then
            echo "tag=beta" >> $GITHUB_OUTPUT
          fi
          if [[ $last_commit_message == *"(release)" ]]; then
            echo "tag=release" >> $GITHUB_OUTPUT
          fi

      - name: Bump version and push tag
        id: tag_version
        if: which_tag.ref == 'refs/heads/main' && steps.which_tag.outputs.tag == 'release'
        uses: mathieudutour/github-tag-action@v6.0
        with:
          github_token: ${{ secrets.GITHUB_TOKEN  }}
      - name: Bump version and push tag
        id: tag_version
        if: which_tag.ref == 'refs/heads/main' && (steps.which_tag.outputs.tag == 'alpha' || steps.which_tag.outputs.tag == 'beta')
        uses: mathieudutour/github-tag-action@v6.0
        with:
          github_token: ${{ secrets.GITHUB_TOKEN  }}
          dry_run: true
      - name: Add tag to output step for main branch
        id: tag
        run: |
          if [ '${{ github.ref }}' = 'refs/heads/main' && steps.which_tag.outputs.tag = 'release' ]; then
            echo "new_version=${{ steps.tag_version.outputs.new_version }}" >> $GITHUB_OUTPUT
          fi
          if [ '${{ github.ref }}' = 'refs/heads/main' && (steps.which_tag.outputs.tag = 'alpha' ]; then
            echo "new_version=${{ steps.tag_version.outputs.new_version }}-alpha${{ github.run_number }}" >> $GITHUB_OUTPUT
          fi
          if [ '${{ github.ref }}' = 'refs/heads/main' && (steps.which_tag.outputs.tag = 'beta' ]; then
            echo "new_version=${{ steps.tag_version.outputs.new_version }}-beta${{ github.run_number }}" >> $GITHUB_OUTPUT
          fi

      - uses: actions/setup-node@v2
        with:
          node-version: 18

      - uses: pnpm/action-setup@v2
        name: Install pnpm
        with:
          version: 8.5.1
          run_install: false
      
      # Oidc Client
      - name: pnpm version ${{ steps.tag_version.outputs.new_tag }}
        if: github.ref == 'refs/heads/main' && (steps.which_tag.outputs.tag == 'release' || steps.which_tag.outputs.tag == 'alpha' || steps.which_tag.outputs.tag == 'beta')
        run: pnpm version ${{ steps.tag_version.outputs.new_tag }}
        working-directory: ./packages/oidc-client

      - name: pnpm install
        run: pnpm i --frozen-lockfile
        working-directory: ./packages/oidc-client

      - name: pnpm prepare
        run: pnpm run prepare
        working-directory: ./packages/oidc-client

      - id: publish-oidc-client
        uses: JS-DevTools/npm-publish@v1
        if: github.ref == 'refs/heads/main' && (steps.which_tag.outputs.tag == 'release' || steps.which_tag.outputs.tag == 'alpha' || steps.which_tag.outputs.tag == 'beta')
        with:
          token: ${{ secrets.NPM_TOKEN }}
          package: ./packages/oidc-client/package.json
      
      # React Oidc
      - name: npm version ${{ steps.tag_version.outputs.new_tag }}
        if: github.ref == 'refs/heads/main' && (steps.which_tag.outputs.tag == 'release' || steps.which_tag.outputs.tag == 'alpha' || steps.which_tag.outputs.tag == 'beta')
        run: npm version ${{ steps.tag_version.outputs.new_tag }}
        working-directory: ./packages/react-oidc

      - name: pnpm i
        run: pnpm i --frozen-lockfile
        working-directory: ./packages/react-oidc

      - name: pnpm prepare
        run: pnpm run prepare
        working-directory: ./packages/react-oidc

      - name: pnpm test
        run: pnpm test -- --run
        working-directory: ./packages/react-oidc

      - id: publish-react
        uses: JS-DevTools/npm-publish@v1
        if: github.ref == 'refs/heads/main' && (steps.which_tag.outputs.tag == 'release' || steps.which_tag.outputs.tag == 'alpha' || steps.which_tag.outputs.tag == 'beta')
        with:
          token: ${{ secrets.NPM_TOKEN }}
          package: ./packages/react-oidc/package.json

      - name: SonarCloud Scan
        uses: sonarsource/sonarcloud-github-action@master
        if: github.event.pull_request.head.repo.full_name == github.repository && !github.event.pull_request.head.repo.fork
        env:
          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
          SONAR_TOKEN: ${{ secrets.SONAR_TOKEN }}
        with:
          args: >
            -Dsonar.organization=axaguildev
            -Dsonar.projectKey=AxaGuilDEv_react-oidc
            -Dsonar.exclusions=**/*.spec.js,**/*.stories.js,Scripts/**,**/*.scss,**/__snapshots__/**,**/*[Tt]ests.cs,**/node_modules/**,**/ClientApp/build/**,**/ClientApp/.storybook/**,**/ClientApp/storybook-static/**,**/obj/**,**/__mocks__/**,**/ClientApp/src/serviceWorker.ts
            -Dsonar.javascript.lcov.reportPaths=**/coverage/lcov.info

      - name: Commit updates package.json
        uses: stefanzweifel/git-auto-commit-action@v4
        if: github.ref == 'refs/heads/main'
        with:
          commit_message: "[skip ci] Update version package.json"
          commit_user_name: GitHub
          commit_user_email: github-action@bot.com
          commit_author: GitHub <github-action@bot.com>
          push_options: '--force'

      - name: Create a GitHub release
        uses: ncipollo/release-action@v1
        if: github.ref == 'refs/heads/main'
        with:
          tag: ${{ steps.tag_version.outputs.new_tag }}
          name: Release ${{ steps.tag_version.outputs.new_tag }}
          body: ${{ steps.tag_version.outputs.changelog }}

```

**Comment by jafin** [comment]
> Here an idea of a solution, continue to use last git commit comment and automate everything.
> 
> * feat(oidc): my message (alpha) => will publish next minor as an alpha
> * feat(oidc): my message (beta) => will publish next minor as an beta
> * feat(oidc): my message (release)  => will publish next minor release (with automatic git tag and release)
> 
> With the script bellow (not tested). It is more intricate, but I think workflow would be simpler.
> 
> What do you think about ?

I don't have a strong opinion, perhaps try it and see how it works.
It sounds a bit like https://github.com/googleapis/release-please  

Also, I have updated the PR with the fixes you highlighted.

**Comment by guillaume-chervet** [comment]
Yes @jafin , let's try like this. 
Do you want i update CI build from another PullRequest and after merge or do you to merge your PullRequest in this state?

**Comment by jafin** [comment]
@guillaume-chervet sure if you can update the CI first, then I'll rebase onto the changes for this PR.  Or feel free to amend it as part of this change.  Whatever is easiest for you.

**Comment by jafin** [comment]
@guillaume-chervet 

Some suggested changes.
- id: must be unique, changed tag_version for release to `tag_release`   (I assume running the tag-action twice is ok?)
- drop condition on tag_version so it runs for all versions (but still with dryrun) 
- missing `steps.` on line 81/86 if condition


```yml
# This workflow will run tests using node and then publish a package to GitHub Packages when a release is created
# For more information see: https://help.github.com/actions/language-and-framework-guides/publishing-nodejs-packages

name: React Oidc CI

on:
  push:
    branches: [ main ]
  pull_request:
    branches: [ main ]

jobs:
  skip_ci:
    runs-on: ubuntu-latest
    outputs:
      canSkip: ${{ steps.check.outputs.canSkip }}
    steps:
      - id: check
        uses: Legorooj/skip-ci@main
  tests:
    runs-on: ubuntu-latest
    if: needs.skip_ci.outputs.canSkip != 'true' && github.event.pull_request.head.repo.fork
    steps:
      - uses: actions/checkout@v3
      - uses: actions/setup-node@v2
        with:
          node-version: 18

      - uses: pnpm/action-setup@v2
        name: Install pnpm
        with:
          version: 8.5.1
          run_install: false

      - name: pnpm i
        run: pnpm i --frozen-lockfile
        working-directory: ./packages/react-oidc

      - name: pnpm prepare
        run: pnpm run prepare
        working-directory: ./packages/react-oidc

      - name: pnpm test
        run: pnpm test -- --run
        working-directory: ./packages/react-oidc

      - name: pnpm install
        run: pnpm i --frozen-lockfile
        working-directory: ./packages/oidc-client

      - name: pnpm prepare
        run: pnpm run prepare
        working-directory: ./packages/oidc-client

  build:
    environment: react-oidc
    runs-on: ubuntu-latest
    if: needs.skip_ci.outputs.canSkip != 'true' && !github.event.pull_request.head.repo.fork
    steps:
      - uses: actions/checkout@v3
        with:
          token: ${{ secrets.GIT_TOKEN }}
      - name: Determine Alpha, Beta or Release
        id: which_tag
        run: |
          last_commit_message=$(git log --format=%B -n 1)
          # Check if last comment and with "(alpha)"
          if [[ $last_commit_message == *"(alpha)" ]]; then
            echo "tag=alpha" >> $GITHUB_OUTPUT
          fi
          if [[ $last_commit_message == *"(beta)" ]]; then
            echo "tag=beta" >> $GITHUB_OUTPUT
          fi
          if [[ $last_commit_message == *"(release)" ]]; then
            echo "tag=release" >> $GITHUB_OUTPUT
          fi

      - name: Bump version and push tag
        id: tag_release
        if: steps.which_tag.ref == 'refs/heads/main' && steps.which_tag.outputs.tag == 'release'
        uses: mathieudutour/github-tag-action@v6.0
        with:
          github_token: ${{ secrets.GITHUB_TOKEN  }}
      - name: Bump version and push tag
        id: tag_version
        uses: mathieudutour/github-tag-action@v6.0
        with:
          github_token: ${{ secrets.GITHUB_TOKEN  }}
          dry_run: true
      - name: Add tag to output step for main branch
        id: tag
        run: |
          if [ '${{ github.ref }}' = 'refs/heads/main' && steps.which_tag.outputs.tag = 'release' ]; then
            echo "new_version=${{ steps.tag_version.outputs.new_version }}" >> $GITHUB_OUTPUT
          fi
          if [ '${{ github.ref }}' = 'refs/heads/main' && (steps.which_tag.outputs.tag = 'alpha' ]; then
            echo "new_version=${{ steps.tag_version.outputs.new_version }}-alpha${{ github.run_number }}" >> $GITHUB_OUTPUT
          fi
          if [ '${{ github.ref }}' = 'refs/heads/main' && (steps.which_tag.outputs.tag = 'beta' ]; then
            echo "new_version=${{ steps.tag_version.outputs.new_version }}-beta${{ github.run_number }}" >> $GITHUB_OUTPUT
          fi

      - uses: actions/setup-node@v2
        with:
          node-version: 18

      - uses: pnpm/action-setup@v2
        name: Install pnpm
        with:
          version: 8.5.1
          run_install: false
      
      # Oidc Client
      - name: pnpm version ${{ steps.tag_version.outputs.new_tag }}
        if: github.ref == 'refs/heads/main' && (steps.which_tag.outputs.tag == 'release' || steps.which_tag.outputs.tag == 'alpha' || steps.which_tag.outputs.tag == 'beta')
        run: pnpm version ${{ steps.tag_version.outputs.new_tag }}
        working-directory: ./packages/oidc-client

      - name: pnpm install
        run: pnpm i --frozen-lockfile
        working-directory: ./packages/oidc-client

      - name: pnpm prepare
        run: pnpm run prepare
        working-directory: ./packages/oidc-client

      - id: publish-oidc-client
        uses: JS-DevTools/npm-publish@v1
        if: github.ref == 'refs/heads/main' && (steps.which_tag.outputs.tag == 'release' || steps.which_tag.outputs.tag == 'alpha' || steps.which_tag.outputs.tag == 'beta')
        with:
          token: ${{ secrets.NPM_TOKEN }}
          package: ./packages/oidc-client/package.json
      
      # React Oidc
      - name: npm version ${{ steps.tag_version.outputs.new_tag }}
        if: github.ref == 'refs/heads/main' && (steps.which_tag.outputs.tag == 'release' || steps.which_tag.outputs.tag == 'alpha' || steps.which_tag.outputs.tag == 'beta')
        run: npm version ${{ steps.tag_version.outputs.new_tag }}
        working-directory: ./packages/react-oidc

      - name: pnpm i
        run: pnpm i --frozen-lockfile
        working-directory: ./packages/react-oidc

      - name: pnpm prepare
        run: pnpm run prepare
        working-directory: ./packages/react-oidc

      - name: pnpm test
        run: pnpm test -- --run
        working-directory: ./packages/react-oidc

      - id: publish-react
        uses: JS-DevTools/npm-publish@v1
        if: github.ref == 'refs/heads/main' && (steps.which_tag.outputs.tag == 'release' || steps.which_tag.outputs.tag == 'alpha' || steps.which_tag.outputs.tag == 'beta')
        with:
          token: ${{ secrets.NPM_TOKEN }}
          package: ./packages/react-oidc/package.json

      - name: SonarCloud Scan
        uses: sonarsource/sonarcloud-github-action@master
        if: github.event.pull_request.head.repo.full_name == github.repository && !github.event.pull_request.head.repo.fork
        env:
          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
          SONAR_TOKEN: ${{ secrets.SONAR_TOKEN }}
        with:
          args: >
            -Dsonar.organization=axaguildev
            -Dsonar.projectKey=AxaGuilDEv_react-oidc
            -Dsonar.exclusions=**/*.spec.js,**/*.stories.js,Scripts/**,**/*.scss,**/__snapshots__/**,**/*[Tt]ests.cs,**/node_modules/**,**/ClientApp/build/**,**/ClientApp/.storybook/**,**/ClientApp/storybook-static/**,**/obj/**,**/__mocks__/**,**/ClientApp/src/serviceWorker.ts
            -Dsonar.javascript.lcov.reportPaths=**/coverage/lcov.info

      - name: Commit updates package.json
        uses: stefanzweifel/git-auto-commit-action@v4
        if: github.ref == 'refs/heads/main'
        with:
          commit_message: "[skip ci] Update version package.json"
          commit_user_name: GitHub
          commit_user_email: github-action@bot.com
          commit_author: GitHub <github-action@bot.com>
          push_options: '--force'

      - name: Create a GitHub release
        uses: ncipollo/release-action@v1
        if: github.ref == 'refs/heads/main'
        with:
          tag: ${{ steps.tag_version.outputs.new_tag }}
          name: Release ${{ steps.tag_version.outputs.new_tag }}
          body: ${{ steps.tag_version.outputs.changelog }}
```

**Comment by guillaume-chervet** [comment]
I merge on main but it does work. Please waite I will continue to debug on main tomorrow morning. Once it will work. I will squash my commit on main to clean it.

**Comment by guillaume-chervet** [comment]
It is OK now, you can update merge your branch with lastest main.

**Review by guillaume-chervet — APPROVED** [review]

**Comment by guillaume-chervet** [comment]
Thank you so much @jafin

**Comment by guillaume-chervet** [comment]
Which kind of merge do you prefer @jafin .
Again thank you so much

**Comment by jafin** [comment]
@guillaume-chervet just squash it I think.

**Pull request merged**